### PR TITLE
Emit delete appmap event reliably

### PR DESCRIPTION
**Pull request by moxw**
Fixes: #833 

feat(deleteAppMap.ts): ensure reliable delete event emission by adding retry on metadata.json removal

### CODE REVIEW

File: src/lib/deleteAppMap.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ export default async function deleteAppMap(
 
   const indexDir = uri.fsPath.substring(0, uri.fsPath.lastIndexOf('.appmap.json'));
 
+  // This triggers the FileSystemWatcher to emit a delete event for the AppMap file reliably.
+  await retry(async () => rm(`${indexDir}/metadata.json`, { force: true }));
```
Comment (dustinbyrne) [inline src/lib/deleteAppMap.ts:24]: This should be handled just below during the removal of `indexDir/*`. Is this causing a different behavior?

Comment (moxw) [reply src/lib/deleteAppMap.ts:24]: Yes, it appears the general deletion wasn't triggering VSCode's delete event, necessitating this specific file removal.

Comment (dustinbyrne) [reply src/lib/deleteAppMap.ts:24]: It's difficult for me to see how this change affects anything. Take a look at [`IndexJanitor`](https://github.com/getappmap/vscode-appland/blob/293316a9bac11d0c74733f4a19a44184080512df/src/lib/indexJanitor.ts) for more details.

Comment (moxw) [reply src/lib/deleteAppMap.ts:24]: @dustinbyrne, I found that deleting metadata.json specifically triggers the VSCode delete event more reliably than removing indexDir/*. It'd be great if you could test this on your end too.

Comment (ahtrotta) [reply src/lib/deleteAppMap.ts]: @dustinbyrne for whatever reason the change that @moxw made does indeed fix the problem :man_shrugging:

**Review by ahtrotta — APPROVED** [review]

**Pull request merged**